subworkflows/local/arriba_workflow/main.nf

```diff
@@ -0,0 +1,94 @@
+        ch_arriba_ref_cytobands         // channel [ meta, path_cytobands     ]
+        ch_arriba_ref_protein_domains   // channel [ meta, path_proteins      ]
+        arriba                          // boolean
+        all                             // boolean
+        fusioninspector_only            // boolean
+        star_ignore_sjdbgtf             // boolean
+        seq_center                      // string
+        arriba_fusions                  // path?
+        cram                            // array?
+
+    main:
+        ch_versions = Channel.empty()
+        ch_dummy_file = file("$baseDir/assets/dummy_file_arriba.txt", checkIfExists: true)
+
+        if (( arriba || all) && !fusioninspector_only) {
+
+            STAR_FOR_ARRIBA(
+                reads,
+                ch_starindex_ref,
+                ch_gtf,
+                star_ignore_sjdbgtf,
+                '',                       // seq_platform, should be a params like other pipelines
+                seq_center
+            )
+
+            ch_versions = ch_versions.mix(STAR_FOR_ARRIBA.out.versions)
+
+            if ( arriba_fusions ) {
+
+                ch_arriba_fusions = reads.combine( Channel.value( file( arriba_fusions, checkIfExists: true ) ) ) // Should this be done in the main script?
+                    .map { meta, reads, fusions -> [ meta, fusions ] }
+                ch_arriba_fusion_fail = ch_dummy_file
+
+            } else {
+
+                ARRIBA_ARRIBA (
+                    STAR_FOR_ARRIBA.out.bam,
+                    ch_fasta,
+                    ch_gtf,
+                    ch_arriba_ref_blacklist.map{ it[1] },       // should we update nf-core module to includes meta?
+                    ch_arriba_ref_known_fusions.map{ it[1] },
+                    ch_arriba_ref_cytobands.map{ it[1] },
+                    ch_arriba_ref_protein_domains.map{ it[1] }
+                )
+
+                ch_versions = ch_versions.mix(ARRIBA_ARRIBA.out.versions)
+
+                ch_arriba_fusions     = ARRIBA_ARRIBA.out.fusions
+                ch_arriba_fusion_fail = ARRIBA_ARRIBA.out.fusions_fail.map{ meta, file -> return file }
+            }
+
+            if ( cram.contains('arriba') ) {
+
+                SAMTOOLS_SORT_FOR_ARRIBA(STAR_FOR_ARRIBA.out.bam, ch_fasta)
+                ch_versions = ch_versions.mix(SAMTOOLS_SORT_FOR_ARRIBA.out.versions )
+
+                SAMTOOLS_VIEW_FOR_ARRIBA(SAMTOOLS_SORT_FOR_ARRIBA.out.bam.map { meta, bam -> [ meta, bam, [] ] }, ch_fasta, [])
+                ch_versions = ch_versions.mix(SAMTOOLS_VIEW_FOR_ARRIBA.out.versions )
+
+                SAMTOOLS_INDEX_FOR_ARRIBA(SAMTOOLS_VIEW_FOR_ARRIBA.out.cram)
+                ch_versions = ch_versions.mix(SAMTOOLS_INDEX_FOR_ARRIBA.out.versions )
+
+            }
+
+        } else {
+            // Not sure how this dummy file can be useful
+            // If this tool can be skipped, why not just emitting an empty channel?
```

Comment:
remove this?

Reply:
Lets also address this on a separate ticket, add string "TODO". We can then have a new ticket for addressing TODOs in the pipeline